R/layer.R

```diff
@@ -430,7 +430,8 @@ Layer <- ggproto("Layer", NULL,
       c(names(data), names(self$aes_params)),
       snake_class(self$geom)
     )
-    self$computed_geom_params <- self$geom$setup_params(data, c(self$geom_params, self$aes_params))
+    params <- defaults(c(self$geom_params, self$aes_params), self$geom$default_params)
+    self$computed_geom_params <- self$geom$setup_params(data, params)
```

Comment:
3. Before we add computed components to the parameters, we initialise the parameters from the new field